### PR TITLE
GG-29247  Introduce commands for command line interface to manage binary types

### PR DESCRIPTION
**Pull request by tledkov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataRemoveTask.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                    BinaryMetadata meta = ((CacheObjectBinaryProcessorImpl)ignite.context().cacheObjects())
+                        .binaryMetadata(typeId);
+
+                    byte[] marshalled = U.marshal(ignite.context(), meta);
+
+                    res = new MetadataMarshalled(marshalled, meta);
+
+                    ignite.context().cacheObjects().removeType(typeId);
+
+                    future = ignite.compute().broadcastAsync(new DropAllThinSessionsJob());;// Drop all connection
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataRemoveTask.java:107]: extra ;

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataRemoveTask.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        private static final long serialVersionUID = 0L;
+
+        /** Auto-inject job context. */
+        @JobContextResource
+        private transient ComputeJobContext jobCtx;
+
+        /** Metadata future. */
+        private transient IgniteFuture<Void> future;
+
+        private transient MetadataMarshalled res;
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataRemoveTask.java:78]: missing javadoc

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataTypeArgs.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    /** Serial version uid. */
+    private static final long serialVersionUID = 0L;
+
+    /** Type name argument. */
+    public static final String OPT_TYPE_NAME = "--typeName";
+
+    /** Type ID argument. */
+    public static final String OPT_TYPE_ID = "--typeId";
+
+    /** Config. */
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataTypeArgs.java:38]: wrong javadoc

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataTypeArgs.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    private String typeName;
+
+    /** Metrics. */
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataTypeArgs.java:41]: and here

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataTypeArgs.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                case OPT_TYPE_ID:
+                    typeId = argIter.nextIntArg("typeId");
+
+                    break;
+            }
+        }
+
+        if (typeName == null && typeId == null) {
+            throw new IllegalArgumentException("Type to remove is not specified. " +
+                "Please add one of the options: --typeName <type_name> or --typeId <type_ID>");
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataTypeArgs.java:118]: type_ID -> type_id

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataUpdateTask.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+            throw new IgniteException("Invalid job results: " + results);
+
+        if (results.get(0).getException() != null)
+            throw results.get(0).getException();
+        else
+            return results.get(0).getData();
+    }
+
+    /**
+     * Job for {@link CheckIndexInlineSizes} command.
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/tasks/MetadataUpdateTask.java:63]: wrong javadoc

File: modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java [hunk 7]
```diff
@@ -1332,6 +1332,9 @@ private void onDisconnect(JdbcThinTcpIo cliIo) {
 
             stmts.clear();
         }
+
+        // Clear local metadata cache on disconnect.
+        metaHnd = new JdbcBinaryMetadataHandler();
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java:1337]: binaryContext still keeps the old handler

File: modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java [hunk 8]
```diff
@@ -252,7 +252,7 @@
     private final BinaryContext ctx;
 
     /** Binary metadata handler. */
-    private final JdbcBinaryMetadataHandler metaHnd;
+    private JdbcBinaryMetadataHandler metaHnd;
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java:255]: Are you sure metaHnd is not invoked in concurrent ?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/CacheObjectBinaryProcessorImpl.java [hunk 9]
```diff
@@ -893,6 +893,28 @@ else if (holder == null || !holder.metadata().hasSchema(schemaId)) {
         });
     }
 
+    /**
+     * @return Custer binary metadata.
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/CacheObjectBinaryProcessorImpl.java:897]: mistyped cluster

File: modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java [hunk 10]
```diff
@@ -361,6 +361,66 @@ private void unregisterMBean() {
...
+            if (log.isInfoEnabled()) {
+                log.info("Client session has been dropped: "
+                    + clientConnectionDescription(ses, connCtx));
+            }
+        }
+    }
+
+    /**
+     * Compose connection description string.
+     * @param ses client NIO session.
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java:387]: Missed dot and capital letters

File: modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java [hunk 11]
```diff
@@ -361,6 +361,66 @@ private void unregisterMBean() {
...
+     * @param ctx client connection context.
+     * @return connection description
+     */
+    @SuppressWarnings("StringConcatenationInsideStringBufferAppend")
+    private static String clientConnectionDescription(
+        GridNioSession ses,
+        ClientListenerConnectionContext ctx)
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java:394]: ) should be on the next line

File: modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java [hunk 12]
```diff
@@ -361,6 +361,66 @@ private void unregisterMBean() {
...
+        InetSocketAddress locAddr = ses.localAddress();
+
+        assert rmtAddr != null;
+        assert locAddr != null;
+
+        String rmtAddrStr = rmtAddr.getHostString() + ":" + rmtAddr.getPort();
+        String locAddrStr = locAddr.getHostString() + ":" + locAddr.getPort();
+
+        sb.append("id=" + ctx.connectionId());
+        sb.append(", user=").append(authCtx == null ? "<anonymous>" : authCtx.userName());
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java:417]: there is securityContext. See `ClientListenerAbstractConnectionContext#initClientDescriptor`

**Review by korlov42 — CHANGES_REQUESTED** [review]
Error messages in argument parsing frustrate me. Let's provide information that helps to solve the originated problem

File: modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java (hunk 8 above)
Comment (tledkov-gridgain) [reply modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java:255]: JDBC Connection is not thread safe by specification.

File: modules/core/src/test/resources/org.apache.ignite.util/GridCommandHandlerClusterByClassTest_help.output [hunk 13]
```diff
@@ -107,6 +107,22 @@ This utility can do the following commands:
   Set new tracing configuration. If both --scope and --label are specified then add or override label specific configuration, if only --scope is specified, then override scope specific configuration. Print applied configuration.
     control.(sh|bat) --tracing-configuration set (--scope DISCOVERY|EXCHANGE|COMMUNICATION|TX [--label] [--sampling-rate Decimal value between 0 and 1, where 0 means never and 1 means always. More or less reflects the probability of sampling specific trace.] [--included-scopes Set of scopes with comma as separator  DISCOVERY|EXCHANGE|COMMUNICATION|TX])
 
+  Prints metadata command help:
```
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/core/src/test/resources/org.apache.ignite.util/GridCommandHandlerClusterByClassTest_help.output]: It appears that this is not relevant for GG CE, relevant for AI only.

**Pull request closed without merging**